src/leaderboard/messages/data/fetch_by_score.rs

```diff
@@ -0,0 +1,184 @@
+use super::{fetch::FetchResponse, Order, RankedElement};
+use crate::leaderboard::MomentoRequest;
+use crate::utils::prep_leaderboard_request_with_timeout;
+use crate::{Leaderboard, MomentoError, MomentoErrorCode, MomentoResult};
+
+use momento_protos::common::Unbounded;
+use momento_protos::leaderboard::score_range::{Max, Min};
+
+use std::ops::Range;
+
+/// Represents a range of scores used to request elements by score.
+pub struct ScoreRange {
+    min: Option<f64>,
+    max: Option<f64>,
+}
+
+impl ScoreRange {
+    /// Constructs a new `ScoreRange`.
+    pub fn new(min: Option<f64>, max: Option<f64>) -> Self {
+        Self { min, max }
+    }
+
+    /// Constructs a new `ScoreRange` with no bounds.
+    pub fn unbounded() -> Self {
+        Self {
+            min: None,
+            max: None,
+        }
+    }
+
+    /// Validates the score range.
+    pub fn validate(&self) -> MomentoResult<()> {
```

Comment:
why is this public? Why is it possible to create an invalid ScoreRange? Shouldn't `new()` return a `MomentoResult<Self>`?

Should `new` just `.clamp` min and max to `f64::MIN` and `f64::MAX` instead of making it an error to write `1.0..`?

Reply:
`1.0..` not working was an oversight my on part. I've added `From<RangeFrom<f64>>` and `From<RangeTo<f64>>` to handle those cases in [0576f7a](https://github.com/momentohq/client-sdk-rust/pull/438/commits/0576f7a470c9655480c712f02611a9d738dbd4f3)

Reply:
The error cases are when the user passes in `min=fp64::POS_INFINITY` or `max=fp64::NEG_INFINITY`. We translate those to the proto `Unbounded` message. That's why the check `val.is_finite()` isn't enough before marshalling to `Unbounded`: if `min` or `max` is the wrong infinity, then it's an error.

Reply:
Well maybe not. The top level methods take as input `impl Into<ScoreRange>`. If we let `new` return `MomentoResult<ScoreRange>` then the callers will have to invoke the function with something like `client.fetch_by_score((1.0..5.0).into()?)`, right? Because the `From<Range<f64>>` implementation is for `ScoreRange`, not `MomentoResult<ScoreRange>`. See j[1c7c02d](https://github.com/momentohq/client-sdk-rust/pull/438/commits/1c7c02d82472e56e412b304d838ec85404c75b42)